packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -524,8 +523,8 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
             unitController={this.props.unitController}
             id='timegraph-chart-1'
             layers={[this.markersChartLayer]}
-        >
-        </ReactTimeGraphContainer>;
+            viewport={true}
```

Comment:
By design, getMarkersContainer() and getChartContainer() MUST return a viewport.

Reply:
> Hmm, I understand the reason now for `viewPort` flag. It exist because the `TimeAxisComponent` and the `TimeNavigatorComponent` are creating a `ReactTimeGraphContainer` that doesn't use view port.
> 
> As a user of `ReactTimeGraphContainer` how would I know when to use `viewPort` to `true` or `false`? I would have to know how things are implemented internally of the timeline-chart library and shouldn't be necessary.

The horizontal and vertical scrollbars, which are also created with a `ReactTimeGraphContainer`, cannot be implemented with a viewport.  There is no horizontal panning, there is no "window" viewing a section of a "world" in those components.  It's just a scrollbar.  (The top axis can be a viewport though!)

This is why the `TimeGraphViewportContainer` class was made, because the methods inside are pertinent to most TimeGraphLayers, but not all.  Also, the moving of stage.position.x needs to be at the Application/Container level, 

Inside of the timeline-chart library it should be clear that TimeGraphViewportLayers need to be added to TimeGraphViewportContainers.  (It could be _more_ clear, but I think it's good enough).

I don't think that the flag is that much of a problem and we can leave it as-is.  However, if you disagree (which is fine)...

**Solutions:**

1) I can add code in `ReactTimeGraphContainer` (or possibly `TimeGraphContainer` in TimelineChart) that can automatically detect whether or not the container should be viewport.  The check: check all layers and see if `layer instanceof TimeGraphViewportLayer`.  This would be the fastest solution.

2)  We can delete `TimeGraphViewportContainer` and move the logic from `TimeGraphViewportContainer` => `TimeGraphViewportLayer`.  That would remove the need for a different TimeGraphContainer (layers have PIXI.Containers anyway), and the layers can shift independently based on the UnitController instead of base PIXI.Container created w/ PIXI.Application.

Reply:
I like simplification gain of the option 2. If you hit a roadblock then we can go with option 0. WDYT?

Reply:
Sounds good to me!  I'll start working on the refactor.